src/internal/connector/graph/concurrency_limiter.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import (
+	"net/http"
+	"sync"
+
+	khttp "github.com/microsoft/kiota-http-go"
+)
+
+// ConcurrencyLimiter middleware limits the number of concurrent requests to graph API.
+// Currently we only enforce concurrency limit for exchange
+type ConcurrencyLimiter struct {
+	semaphoreExch chan struct{}
```

Comment:
Corso can only run a single service at a time.  I don't think you need to namespace exchange for this case.  Lets name it `semaphore`, and we can namespace further as refinements arrive.

Reply:
fair point. will make the change